rust-toolchain.toml

```diff
@@ -0,0 +1,8 @@
+# This file pins the toolchain  so that ./scripts/regenerate_test_rustdocs.sh
+# always produces the desired format. Every new nightly release, if the rustdoc
+# format is unchanged, this toolchain may be incremented. Otherwise we need a new
+# version of the adapter.
+
+[toolchain]
+channel = "nightly-2024-01-09"
```

Comment:
I'd maybe pin this to `nightly`, just to make sure we notice when a new format gets released and so we make sure things still work on that format -- e.g. in case of rustdoc bugs or bugfixes that don't change the format version. Those aren't super common, but they have happened and it's nice to catch them quickly.

Reply:
Fixed.